flexible_event_config.md

```diff
@@ -169,6 +170,15 @@ When the `event_report_window` for a spec completes, we will map its summary val
   "trigger_summary_bucket": [<bucket start>, <bucket end>]
 }
 ```
+### Trigger prioritization
+
+Given that triggering attribution can affect a source's state without producing a report, we will need a new algorithm for doing trigger prioritization. Here is a sketch of how it could work:
+
+1. Always maintain a sorted list of triggers, sorted in order of priority (descending) and trigger time (ascending)
+2. Whenver a spec's window's end time is hit (breaking ties arbitrarily)
```

Comment:
When a new trigger "matching" this source, we will perform the following process. Not at end of a reporting window

Reply:
Why does this matter?

Reply:
I intentionally described this per reporting window to minimize the amount of state / accounting. I think the end result should be the same?

Reply:
To me it's a lot easier to understand this algorithm when it's described as happening at the end of each report window, not when a trigger is handled.

Reply:
@csharrison @giladbarkan-github , no. If process at end of the reporting window, the trigger data with eariler reporting window will have the highest priority.

Reply:
@yanzhangnyc The PR says "iterate through triggers in order". I take that to mean all triggers associated with this source.

Reply:
> @yanzhangnyc The PR says "iterate through triggers in order". I take that to mean all triggers associated with this source.

If so, we need to emphasize "at the end window of any trigger data" instead of just simply mention "trigger window end".

Reply:
I updated the wording. PTAL